jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/StandardDescriptorProcessor.java

```diff
@@ -1028,69 +1038,134 @@ public void addWelcomeFiles(WebAppContext context, XmlParser.Node node, Descript
         }
     }
 
+    /**
+     * Check if a path is eligible to be mapped to the servlet.
+     * A path can be (re)mapped iff:
+     * <ul>
+     *     <li>it is not already mapped</li>
+     *     <li>it is already mapped by webdefault.xml</li>
+     *     <li>it is already mapped by the embedded api, and the descriptor is not webdefault.xml</li>
+     * </ul>
+     * The effect of the above conditions is to produce the following precedence hierarchy:
+     * <ol>
+     *     <li>jetty-override.xml (OverrideDescriptor)</li>
+     *     <li>web.xml (WebDescriptor)</li>
+     *     <li>web-fragment.xml (FragmentDescriptor)</li>
+     *     <li>embedded api</li>
+     *     <li>webdefault.xml (DefaultsDescriptor)</li>
+     * </ol>
+     * @param servletName the servlet target of the mapping
+     * @param path the path to check
+     * @param context the context of the mapping
+     * @param descriptor the descriptor currently being parsed
+     * @return true if the path can be (re)mapped, false otherwise
+     */
+    private boolean checkServletMappingPath(String servletName, String path, WebAppContext context, Descriptor descriptor)
+    {
+        Objects.requireNonNull(servletName);
+        Objects.requireNonNull(path);
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(descriptor);
+
+        ListIterator<ServletMapping> listItor = _servletMappings.listIterator();
+        boolean found = false;
+        while (listItor.hasNext() && !found)
+        {
+            ServletMapping originalMapping = listItor.next();
+            String[] originalPathSpecs = originalMapping.getPathSpecs();
+            if (originalPathSpecs == null || (originalPathSpecs.length == 0))
+                continue;
+
+            for (String originalPath : originalPathSpecs)
+            {
+                if (!originalPath.equals(path))
+                    continue;
+
+                if (servletName.equals(originalMapping.getServletName()))
+                {
+                   //Same path mapped to same servlet, ignore
+                   return false;
```

Comment:
How can we return false here?  why is this within the path spec loop when it doesn't check path specs?